### DISCUSSION TITLE
navigateTo(), abortNavigation() doesn't work on server side

### DISCUSSION THREAD

**Discussion by Harkonnen13** [nuxt/nuxt#17961] [Questions]
Hello. I am trying to implement route middleware, and have some issues with that.

```
export default defineNuxtRouteMiddleware(async (to) => {
    const { fetchList } = useList();
    const { start, finish, fail } = useProgress();

    try {
        const {
            query: { page },
        } = to;

        const currentPage = page ? Number(page) : null;

        start();
        const res = await fetchList(currentPage);
        finish();
        console.log(res, 'finish');
    } catch (error) {
        fail();
        if (process.server) return navigateTo('/');
        else return abortNavigation();
    }
});
```

So, the point is - i am loading data list, and accepting the route only if fetch is complete.
If loading fails, on server side it should redirect to main page, on client side - just abort navigation.

I thought middlewares should be isomorthic and works same way on server and client sides, but when i call "navigateTo('/')" on server side, i see this: 

![image](https://user-images.githubusercontent.com/28978908/152785743-6ae3fc15-12a6-4b3d-bac1-ca3700149c34.png)

on client side it works perfectly.
Any thoughts?

**Comment by wobsoriano** [comment]
I think the problem is in your `try` block not having a `return`?

```
Possible return values are:

nothing - does not block navigation and will move to the next middleware function,
if any, or complete the route navigation
```

**Reply by Harkonnen13** [reply]
@wobsoriano Thank you for ur answer, but the problem is in function navigateTo
On client side it works
https://github.com/nuxt/framework/issues/3105
